godot-core/src/obj/phantom_var.rs

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::{cmp::Ordering, fmt, hash::Hash, marker::PhantomData};
+
+use crate::{
+    meta::{ClassName, GodotConvert, GodotType, PropertyHintInfo},
+    registry::property::{Export, Var},
+};
+
+/// A zero-sized type for creating a property without a backing field, accessible only through a getter and/or setter function.
+///
+/// This must be used in a struct deriving [`GodotClass`](../register/derive.GodotClass.html) and requires that the field has
+/// an explicit [`#[var]` attribute](../register/derive.GodotClass.html#register-properties--var) and either a custom getter,
+/// a custom setter, or both.
+///
+/// The generic type `T` is the type of the property, and must itself implement [`Var`].
+/// `T` is also the type returned by the getter and accepted by the setter.
+///
+/// Like `PhantomData` from the Rust standard library, `PhantomVar` implements many common traits like `Eq` and `Hash`
```

Comment:
Why do I need this info? Docs will inform me about all the auto-implementations anyway.